src/framework/ui/uitypes.h

```diff
@@ -187,9 +187,32 @@ enum class Checkable {
     Yes
 };
 
+enum class ActionCategory {
+    Undefined = -1,
+    Internal,
```

Comment:
It looks like the `Internal` category is not really used; its role already seems to be fulfilled by `Undefined`. Do we need it?